src/main/kotlin/com/hibob/academy/resource/OwnerApi.kt

```diff
@@ -0,0 +1,86 @@
+package com.hibob.academy.resource
+
+import com.hibob.restAPI.Owner
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+import java.util.concurrent.CopyOnWriteArrayList
+
+@Controller
+@Path("/api/adi/owners")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+class OwnerResource {
+
+    private val allOwners: MutableList<Owner> = CopyOnWriteArrayList()
+
+    @GET
+    @Path("/{ownerId}")
+    fun getOwnerById(@PathParam("ownerId") ownerId: Long): Response {
+        val owner = allOwners.find { it.id == ownerId.toInt() }
+        return if (owner != null)
+            Response.ok(owner).build()
+        else
+            Response.status(Response.Status.NOT_FOUND)
+                .entity("Owner not found")
+                .build()
+    }
+
+    @GET
+    fun getAllOwners(): Response {
+        return Response.ok(allOwners).build()
+    }
+
+    @POST
+    fun addOwner(owner: Owner): Response {
+        val newOwnerId = (allOwners.maxOfOrNull { it.id } ?: 0) + 1
+
+        val badRequest = Response.status(Response.Status.BAD_REQUEST).build()
+        val firstName = owner.name?.split(" ")?.first() ?: owner.firstName ?: return badRequest
+        val lastName = owner.name?.split(" ")?.last() ?: owner.lastName ?: return badRequest
```

Comment:
I like that you handled the missing names with a `BAD_REQUEST`. It can be a bit more informative if you add `Response.status(Response.Status.BAD_REQUEST).entity("Missing last name").build(), so that the user can know what field they are missing in this case.

Also `last` will return the last element of the array. So if someone is called Jamie Van Halen then the last name won't be Van Halen in your case, only Halen which could be wrong.